web/web.go

```diff
@@ -582,25 +583,60 @@ func (h *Handler) rules(w http.ResponseWriter, r *http.Request) {
 	h.executeTemplate(w, "rules.html", h.ruleManager)
 }
 
-func (h *Handler) targets(w http.ResponseWriter, r *http.Request) {
-	// Bucket targets by job label
+// Bucket targets by job label
+func bucketTargets(targets []*retrieval.Target) struct {
+	TargetPools map[string][]*retrieval.Target
+} {
 	tps := map[string][]*retrieval.Target{}
-	for _, t := range h.targetManager.Targets() {
+	for _, t := range targets {
 		job := t.Labels().Get(model.JobLabel)
+		if job == "" {
+			job = t.DiscoveredLabels().Get(model.JobLabel)
```

Comment:
How did this work in the targets page before?

Reply:
I added this bit of logic to account for dropped targets so they would be included in the bucketing when displaying the service discovery page. It has no use for the targets page. 

Do you think it would be better practice for the target and service discovery pages to have their own bucketing rather than include this if branch?

Reply:
Yep, that should be okay. Showing dropped targets in the targets page is technically wrong.

Reply:
For the SD page it should be bucketed by the scrape_config job_name, not the labels of the targets.

Reply:
I've simplified this part of the code now bucketing by the job name of the targetSets.